_includes/app/txn-sample.cpp

```diff
@@ -0,0 +1,76 @@
+// Build with g++ -std=c++11 txn_sample.cpp -lpq -lpqxx
+
+#include <cassert>
+#include <functional>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <pqxx/pqxx>
+
+using namespace std;
+
+void transferFunds(
+    pqxx::dbtransaction *tx, int from, int to, float amount) {
+  // Read the balance.
+  pqxx::result r = tx->exec(
+      "SELECT balance FROM accounts WHERE id = " + to_string(from));
+  assert(r.size() == 1);
+  float fromBalance = r[0][0].as<float>();
+
+  if (fromBalance < amount) {
+    throw domain_error("insufficient funds");
+  }
+
+  // Perform the transfer.
+  tx->exec("UPDATE accounts SET balance = balance - " 
+      + to_string(amount) + "::DECIMAL WHERE id = " + to_string(from));
+  tx->exec("UPDATE accounts SET balance = balance + " 
+      + to_string(amount) + "::DECIMAL WHERE id = " + to_string(to));
+}
+
+// txnWrapper runs fn inside a transaction and retries it as needed.
+void txnWrapper(
+    pqxx::connection *c, function<void (pqxx::dbtransaction *tx)> fn) {
+  pqxx::work tx(*c);
+  while (true) {
+    try {
+      pqxx::subtransaction s(tx, "cockroach_restart");
+      fn(&s);
+      s.commit();
+      break;
+    } catch (const pqxx::pqxx_exception& e) {
+      // Swallow "transaction restart" errors; the transaction will be retried.
+      // Unfortunately libpqxx doesn't give us access to the error code, so we
+      // do string matching to identify retriable errors.
```

Comment:
s/retriable/retryable/ ...?  Neither is correct I guess, but `retryable` wins the Google battle 2 to 1.